controllers/topicPage/TopicPage.js

```diff
@@ -18,43 +18,70 @@ class TopicPage {
     this.getTopicPageById = this.getTopicPageById.bind(this);
   }
 
+  async getFeedTopicPages(payload) {
+    const topicPages = await this._getStreamService.getTopicPages(
+      payload.id,
+      payload.limit,
+      payload.offset
+    );
+    let newTopicPagesWithBlock = new BlockServices(
+      payload.listBlockUser,
+      payload.listBlockDomain,
+      payload.listPostAnonymous
+    ).getHasBlock(topicPages);
+    let results = await filterFeeds(payload.userId, newTopicPagesWithBlock, payload.id);
+    return results;
+  }
+
+  async modifiedFeed(userId, post) {
+    if (post.verb === POST_VERB_POLL) {
+      let postPoll = await modifyPollPostObject(userId, post);
+      return postPoll;
+    } else {
+      return post;
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `modifiedFeed` method is correctly implemented. It checks if the post verb is "poll" and modifies the post object accordingly. If not, it simply returns the original post.



<!-- This is an auto-generated comment by OSS CodeRabbit -->